styx-client/src/main/java/com/spotify/styx/client/auth/GoogleIdTokenAuth.java

```diff
@@ -41,19 +46,32 @@
 public class GoogleIdTokenAuth {
   private static final Logger log = LoggerFactory.getLogger(GoogleIdTokenAuth.class);
   private static final JsonFactory JSON_FACTORY = Utils.getDefaultJsonFactory();
+  private static final String DEFAULT_GCE_METADATA_HOST = "http://169.254.169.254";
+  private static final String GCE_METADATA_IDENTITY_PATH =
+      "/computeMetadata/v1/instance/service-accounts/default/identity{?audience,format}";
 
   private final HttpTransport httpTransport;
   private final Optional<GoogleCredential> credential;
+  private final boolean isGCE;
 
   GoogleIdTokenAuth(
```

Comment:
While you're here... how about making this one public? Or maybe it's one layer above, in the client setup. Basically, it would be cool if I could pass my own credentials (or `ImpersonatedCredential`) when creating the styx client.

Reply:
I'd like to do this in a separate PR, along with switching from `GoogleCredential` to `GoogleCredentials` along with exposing `StyxClientFactory` methods that allow passing in credentials.